server/handler/query.go

```diff
@@ -111,23 +113,21 @@ func Query(db *sql.DB) RequestProcessFunc {
 						colData[columnNames[i]] = sqlVal.String
 					}
 				case *[]byte:
-					// TODO (carlosms) this may not be an array always
-					var protobufs [][]byte
-					if err := json.Unmarshal(*val.(*[]byte), &protobufs); err != nil {
-						return nil, err
-					}
-
-					nodes := make([]*uast.Node, len(protobufs))
-
-					for i, v := range protobufs {
-						node := uast.NewNode()
-						if err = node.Unmarshal(v); err != nil {
+					// TODO (carlosms) this may not always be a JSON array, it could
+					// be a JSON object
+
+					// DatabaseTypeName JSON is used for arrays, but we don't know the
+					// type of each element. We try with uast node first and text later
+					nodes, err := unmarshallUAST(val)
+					if err == nil {
+						colData[columnNames[i]] = nodes
+					} else {
+						var strings []string
```

Comment:
why strings? Maybe `[]interface{}` is better?

Reply:
I did it thinking in the commits.parents column, which is an array of strings. But it actually doesn't even have to be an array, using `interface{}` will work for any JSON array or object we receive. Changed in 28fc331.